### PR TITLE
Update free-programming-books-langs.md

### PR DESCRIPTION
**Pull request by bryankimani**
## What does this PR do?
Add resource(s) 

### Why is this valuable?
A Brief Introduction to Ruby ebook  is a collection of code playbacks designed to help guide a reader through code examples
N+1 Queries Basics ebook takes through the the tools that can help you detect n+1 queries before they hit production.

### How do we know it's really free?
I have went through thee books before and also, they are from trusted free use sources.

### For book lists, is it a book? For course lists, is it a course? etc.

A Brief Introduction to Ruby ebook  is a collection of code playbacks designed to help guide a reader through code examples
N+1 Queries Basics ebook takes through the the tools that can help you detect n+1 queries before they hit production.

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -2256,6 +2257,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Kestrels, Quirky Birds, and Hopeless Egocentricity](https://leanpub.com/combinators/read) - Reg Braithwaite
 * [Learn Ruby on Rails: Book One](https://learn-rails.com/content/learnrails1) - Daniel Kehoe
 * [Learn Ruby on Rails: Book Two](https://learn-rails.com/content/learnrails2) - Daniel Kehoe
+* [N+1 Queries Basics](https://bhserna.com/n-plus-1-queries-basics.html) - Benito Serna
```
Comment (eshellman) [inline books/free-programming-books-langs.md:2260]: in general, we don't list books that require email addresses

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -2205,6 +2205,7 @@ Books on general-purpose programming that don't focus on a specific language are
 
 ### Ruby
 
+* [A Brief Introduction to Ruby](https://markm208.github.io/rubybook) - Mark Mahoney
```
Comment (eshellman) [inline books/free-programming-books-langs.md:2208]: this should go in the interactive tutorials list

**Review by eshellman — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (bryankimani) [reply books/free-programming-books-langs.md:2260]: Thank @eshellman. Noted

File: books/free-programming-books-langs.md (hunk 2 above)
Comment (bryankimani) [reply books/free-programming-books-langs.md:2208]: I am making the change as requested.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by eshellman** [comment]
accidentally closed

**Pull request merged**